lib/open_api_spex/cast/any_of.ex

```diff
@@ -0,0 +1,42 @@
+defmodule OpenApiSpex.Cast.AnyOf do
+  @moduledoc false
+  alias OpenApiSpex.Cast
+
+  def cast(ctx, failed_schemas \\ [])
+
+  def cast(%_{schema: %{type: _, anyOf: []}} = ctx, failed_schemas) do
+    Cast.error(ctx, {:any_of, error_message(failed_schemas)})
+  end
+
+  def cast(
+        %{schema: %{type: _, anyOf: [schema | schemas]}} = ctx,
+        failed_schemas
+      ) do
+    with {:ok, value} <- Cast.cast(%{ctx | schema: schema}) do
+      {:ok, value}
+    else
+      _ ->
+        new_schema = Map.put(ctx.schema, :anyOf, schemas)
```

Comment:
Here too with the use of `Map` on a struct. The `%{struct | atom: value}` syntax is used very frequently throughout this codebase (like the next line, for instance), and it would be good to keep things consistent.